oeps/best-practices/oep-0067-bp-tools-and-technology.rst

```diff
@@ -0,0 +1,343 @@
+   * - Arbiter
+     - Braden MacDonald  <braden@opencraft.com>
+   * - Status
+     - Accepted
+   * - Type
+     - Best Practice
+   * - Created
+     - 2023-09-05
+   * - Review Period
+     - 2023-09-05 - 2023-10-10
+   * - Resolution
+     - * https://github.com/openedx/open-edx-proposals/pull/518
+   * - References
+     - :doc:`oep-0011-bp-FED-technology`
+
+Abstract
+********
+
+The Open edX Platform is built using many 1st and 3rd party tools and technologies.
+This document covers the current recommendations for which tools and technologies should be used.
+
+Motivation
+**********
+
+The Open edX platform has a lot of 1st and 3rd party tools and technology that it uses.
+The problems we're trying to solve are:
+
+    * It is difficult to know what the current best practice for a given area.
+
+    * It is unclear how to go about the discussion and decision making process
+      to change a best practice.
+
+:doc:`OEP-11 <oep-0011-bp-FED-technology>` serves as a great example for the
+usefulness of this kind of OEP but limits itself to be relevant only to
+frontend technologies.  This OEP would supersede :doc:`OEP-11
+<oep-0011-bp-FED-technology>` and encompass not only frontend technolog but any
+technology that:
+
+    * Is considered a best practice for the Open edX Platform
+
+    * Is relevant to a large subset of the Open edX Platform
+
+Specification
+*************
+
+.. note:: We'll fill this out before the OEP lands but leaving mostly blank for initial feedback.
+
+General Technology Selection
+============================
+
+This technology is not specific to frontend or backend code.
+
+#. **Codecov should be used to report code coverage by unit tests**
+
+   **Rationale**: It is important to measure the amount of code covered by our
+   automated test suites. By striving for a high level of test coverage, we can
+   reduce the number of bugs that can only be found via manual testing, 
+   and by using codecov to run coverage in CI,
+   contributors are automatically reminded to include tests for any new code.
+
+   The Open edX community has standardized around `Codecov`_ as a coverage reporting service.
+
+#. **Use Github actions for CI Testing**
+
+   **Rationale**: As a healthy open source project we need good continouse integration
+   testing to prevent the introduction of obvious bugs into the platform code.  The Github Actions
+   tool should be used to perform Continuous Integration(CI) testing for all Open edX repositories
+   that perform any sort of software testing.
+
+Frontend Technology Selection
+=============================
+
+.. note:: 
+
+   The phrase “frontend” is used to mean any part of the platform that is shown to
+   users. This encompasses views rendered in Python on the server, interactive
+   interfaces written using JavaScript, and CSS styling.
+   
+.. _Use React and Redux:
+
+#. **Use React**
+
+   **Rationale**: React must be used for building new UIs, as it is
```

Comment:
Is the guidance really MUST on React? Some UIs are going to be so simple that react is overkill. (See above regarding: "new around here"; if this debate has already been had and resolved then I don't need anybody to restart it for me. It's more a question about unexamined assumptions.)

Reply:
@deborahgu If a UI isn't using React, at least one possible concern is the compatibility with using the Paragon design system and React-based UI component library in terms of brand consistency, etc. across UI projects within Open edX.

Reply:
yeah, this is a bigger question than can be resolved in this OEP, but eg. right now we're working on a place where we're dealing with the entire Paragon dependency stack to render 100% static pages, and it's not lightweight. It might be fundamentally worth the cost, but the cost is not small.